### PR TITLE
[`Trainer` / `GC`] Add `gradient_checkpointing_kwargs` in trainer and training arguments

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Partially fixes: https://github.com/huggingface/trl/pull/912

Following https://github.com/huggingface/transformers/pull/27020 it is important to propagate `gradient_checkpointing_kwargs` in `Trainer` as well

cc @ArthurZucker

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1119,6 +1121,12 @@ class TrainingArguments:
             "help": "If True, use gradient checkpointing to save memory at the expense of slower backward pass."
         },
     )
+    gradient_checkpointing_kwargs: dict = field(
+        default=None,
+        metadata={
+            "help": "Gradient checkpointing key word arguments. Will be passed to `torch.utils.checkpoint.checkpoint` through `model.gradient_checkpointing_enable`."
```
Comment (ArthurZucker) [inline src/transformers/training_args.py:1127]: could you add why anyone would want to pass antyhing and what is supported?

**Review by ArthurZucker — APPROVED** [review]
Thank 🔥

File: src/transformers/training_args.py (hunk 1 above)
Comment (ArthurZucker) [reply src/transformers/training_args.py:1127]: And a small small test!

**Comment by ArthurZucker** [comment]
Also why does it *partially* fix the issue

File: src/transformers/training_args.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/training_args.py]: Done!

**Comment by younesbelkada** [comment]
It partially fixes the issue because I need https://github.com/huggingface/peft/pull/1036 to be merged to fix the bug with respect to PEFT models

**Pull request merged**